internal/docker/docker_impl.go

```diff
@@ -358,6 +358,69 @@ loop:
 	return err
 }
 
+func (d *dockerV20Container) writeFile(path string, content []byte) error {
+	if d.config.Execution.Mode == config.DockerExecutionModeSession && d.config.Execution.DisableAgent {
```

Comment:
Does writing files work if the agent is disabled in connection mode?

Reply:
I'm not sure how that condition made it in there. Fixed.